paddle/fluid/operators/merge_selected_var_op.cc

```diff
@@ -0,0 +1,160 @@
+/* Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "paddle/fluid/framework/op_registry.h"
+#include "paddle/fluid/memory/memcpy.h"
+
+namespace paddle {
+namespace operators {
+
+class MergeSelectedVarOp : public framework::OperatorBase {
+ public:
+  MergeSelectedVarOp(const std::string &type,
+                     const framework::VariableNameMap &inputs,
+                     const framework::VariableNameMap &outputs,
+                     const framework::AttributeMap &attrs)
+      : OperatorBase(type, inputs, outputs, attrs) {}
+
+ private:
+  void RunImpl(const framework::Scope &scope,
+               const platform::Place &dev_place) const override {
```

Comment:
The implementation of `MergeSelectedVarOp` and `SplitSelectedVarOp` is nearly the same. Try to simplify the code (also consider the code of `AssignOp`. Some codes can be reused).

Reply:
Done. 

Create a file named "select_op_helper.h" and put GetBranchNumber which converts Mask tensor to int.

Moving AssignFunctor from assign_op.cc to assign_op.h and use it by including assign_op.h